pipelines/tekton/aiedge-e2e/supported-model-formats.json

```diff
@@ -0,0 +1,30 @@
+[
+    {
+        "model-format": "tensorflow",
+        "model-server": "ovms"
+    },
+    {
+        "model-format": "tensorflow-lite",
+        "model-server": "ovms"
+    },
+    {
+        "model-format": "onnx",
+        "model-server": "ovms"
+    },
+    {
+        "model-format": "pytorch",
+        "model-server": "ovms"
+    },
+    {
+        "model-format": "openvino-ir",
+        "model-server": "ovms"
+    },
+    {
+        "model-format": "scikit-learn",
```

Comment:
Should this be `joblib`? or `scikit-learn-joblib` or something? https://docs.seldon.io/projects/seldon-core/en/latest/servers/sklearn.html

Also, if I understand things correctly, then scikit-learn models can also be stored in mlflow format, so if we listed them as separate options, which should a user choose?

Reply:
> Should this be `joblib`? or `scikit-learn-joblib` or something? https://docs.seldon.io/projects/seldon-core/en/latest/servers/sklearn.html
> 

You may want to check this sources of KServe project: https://github.com/kserve/kserve/tree/8357567e8644840613bb31ec37f5813879557c41/config/runtimes. Look under `supportedModelFormats` (e.g. [sklearn](https://github.com/kserve/kserve/blob/8357567e8644840613bb31ec37f5813879557c41/config/runtimes/kserve-sklearnserver.yaml#L9-L13))

We've have adopted KServe for non-edge cases. In ODH fork, we don't officially support any of those runtimes, but perhaps we may want naming consistency, since there are plans to support only a few of those servers. If we use same naming, this should be `sklearn`.

Notice that in KServe they use two fields: format name, and format version. The version distinction seems important for some formats (e.g. [tensorflow](https://www.tensorflow.org/hub/model_formats) has TF1, TF2, TF.js, TF-Lite). I don't think we need to use two fields here, but we need to make the distinction (e.g. you already used `model-format: tensorflow-lite` in an entry).

> Also, if I understand things correctly, then scikit-learn models can also be stored in mlflow format, so if we listed them as separate options, which should a user choose? This stuff confuses me.

I think this is similar to the other question and answer would be the same: user would pick based on the needed/preferred model server features.

Reply:
> In ODH fork, we don't officially support any of those runtimes, but perhaps we may want naming consistency, since there are plans to support only a few of those servers.

I don't fully understand this bit: I guess you're saying that using KServe in ODH & downstream, we don't support all of the model formats that KServe supports? How does a user determine whether a particular format is supported in ODH or not? What about for OpenVino & Seldon, are there particular model formats that we should or shouldn't support for these for these edge use cases?

Reply:
AFAIK, in KServe we don't support specific formats. What we support are the Model Servers (aka Serving Runtimes). To get support, users would need to save their models in a format that the supported model servers can load.

In ODH, at the moment, the only two Model Servers we formally support are OpenVINO and Caikit/TGIS. 

That said, KServe/ModelMesh are agnostic to formats and servers. Because of the limited number of servers we currently support (and, in transition, the small set of formats), I've heard some users registering their preferred servers into KServe/ModelMesh to use them. We won't support their model server, but we will help with the rest of the stack.

Not sure how much this situation will change, given the edge project seems to need the other model servers.  Minimally, I see that need of picking Seldon.

Reply:
Thanks for the info @israel-hdez.

> In ODH, at the moment, the only two Model Servers we formally support are OpenVINO and Caikit/TGIS.

I'm not familiar with Caikit/TGIS: is that something that would be appropriate for edge, and if so, is there some appropriate example model that we could have here?

> Not sure how much this situation will change, given the edge project seems to need the other model servers. Minimally, I see that need of picking Seldon.

That's something that we should bottom out: are you saying the edge project seems to need Seldon just because we've got examples that use it, or do you think it's a model server that users will explicitly want to use in edge environments?